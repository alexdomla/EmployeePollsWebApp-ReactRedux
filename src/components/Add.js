import React from "react";
import { connect } from "react-redux";
import { useState } from "react";
import { handleAddQuestion } from "../actions/questions";
import { useNavigate } from "react-router-dom";
import Nav from "./Nav";

// Add component, which allows users to add a new question
const Add = ({ dispatch, authedUser, isLoggedIn }) => {
  const navigate = useNavigate();
  const [firstOption, setFirstOption] = useState("");
  const [secondOption, setSecondOption] = useState("");

  // Handle changes in the first option input
  const handleFirstOptionChange = (e) => {
    const value = e.target.value;
    setFirstOption(value);
  };

  // Handle changes in the second option input
  const handleSecondOptionChange = (e) => {
    const value = e.target.value;
    setSecondOption(value);
  };

  // Handle form submit
  const handleSubmit = (e) => {
    e.preventDefault();
    if (firstOption && secondOption) {
      dispatch(handleAddQuestion(firstOption, secondOption));
      navigate("/");
    } else {
      alert("Please fill both options before submitting");
    }
  };

  // Render the form for adding a new question
  return (
    <div className="newPage">
      {authedUser ? <Nav></Nav> : null}
      <h1>Create Poll</h1>
      <p>Would you rather</p>
      <form onSubmit={handleSubmit}>
        <label htmlFor="">First Option</label>
        <input
          type="text"
          id="fname"
          name="fname"
          value={firstOption}
          onChange={handleFirstOptionChange}
        ></input>
        <label htmlFor="">Second Option</label>
        <input
          type="text"
          id="fname"
          name="fname"
          value={secondOption}
          onChange={handleSecondOptionChange}
        ></input>
        <button type="submit">Submit</button>
      </form>
    </div>
  );
};

// Map the authedUser state to props
const mapStateToProps = ({ authedUser }) => ({
  authedUser,
  isLoggedIn: !!authedUser,
});

// Connect the component to the Redux store
export default connect(mapStateToProps)(Add);
