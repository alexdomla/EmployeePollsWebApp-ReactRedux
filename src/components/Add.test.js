import { store } from "../store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Add from "../components/Add";

describe("add", () => {
  it("will match snapshot", () => {
    const component = render(
      <MemoryRouter>
        <Provider store={store}>
          <Add />
        </Provider>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});
