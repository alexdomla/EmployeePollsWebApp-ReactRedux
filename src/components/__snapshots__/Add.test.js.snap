// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="newPage"
      >
        <h1>
          Create Poll
        </h1>
        <p>
          Would you rather
        </p>
        <form>
          <label
            for=""
          >
            First Option
          </label>
          <input
            id="fname"
            name="fname"
            type="text"
            value=""
          />
          <label
            for=""
          >
            Second Option
          </label>
          <input
            id="fname"
            name="fname"
            type="text"
            value=""
          />
          <button
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="newPage"
    >
      <h1>
        Create Poll
      </h1>
      <p>
        Would you rather
      </p>
      <form>
        <label
          for=""
        >
          First Option
        </label>
        <input
          id="fname"
          name="fname"
          type="text"
          value=""
        />
        <label
          for=""
        >
          Second Option
        </label>
        <input
          id="fname"
          name="fname"
          type="text"
          value=""
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
